### PR TITLE
add texsymbols_str

### PR DESCRIPTION
**Pull request by Suavesito-Olimpiada**
This adds a new API function `char *texsymbols_str()`, which return a formatted list of `command:symbol;` which need to be freed with `texfree`.

I added this functionality to be able to call it as a library in a higher-level language (from [LibTeXPrintf.jl](https://github.com/Suavesito-Olimpiada/LibTeXPrintf.jl)), because it is tricky and unreliable to capture `stdout` output. 
cc @stevengj

Also added a small test script `testtexsymbols.sh` that check the first two symbols, in principle I would like to check all of them, but if the order or some are added/deleted it won't work.

### CODE REVIEW

**Comment by bartp5** [comment]
I think we should check all symbols. I do not expect we will often change the symbol table. If we do it should be deliberate and we can generate a new reference table for the test.

File: src/lexer.c [hunk 1]
```diff
@@ -52,6 +52,39 @@ void ListSymbols()
...
+	char *line=calloc(0, 1);
+	p=Unicode2Utf8(0x25CC); /*dotted circle for combining marks */
+	j=0;
+	while(Symbols[j].name)
+	{
+		str=Unicode2Utf8(Symbols[j].unicode);
+		if(IsCombiningMark(Symbols[j].unicode))
+		{
+			slen=(size_t)snprintf(NULL, 0, "%s %s:%s%s;", Symbols[j].name, p, str, p);
+			line=realloc(line, ln+slen+1);
```
Comment (stevengj) [inline src/lexer.c:72]: Repeatedly calling `realloc` like this (as opposed to iteratively doubling the size) is an O(n^2) algorithm, no?

**Comment by stevengj** [comment]
Seems like it would be nicer to simply export the `Symbols` array?

File: src/lexer.c (hunk 1 above)
Comment (Suavesito-Olimpiada) [reply src/lexer.c:72]: Yes, you're right (and I was thinking to avoid `strlen` in for the same reason 😅).

**Comment by Suavesito-Olimpiada** [comment]
I thought about it, but I want to have the same rendering from here without needing to change something in the Julia side if the lib is updated.

**Comment by stevengj** [comment]
> I thought about it, but I want to have the same rendering of symbols from the lib

It's just a table mapping LaTeX names to unicode codepoints.  Why does user need it in a particular format?

(Julia has [an even bigger table](https://github.com/JuliaLang/julia/blob/master/stdlib/REPL/src/latex_symbols.jl) that it might be useful to crib a few more symbols from here.)

File: src/lexer.c (hunk 1 above)
Comment (Suavesito-Olimpiada) [reply src/lexer.c:74]: Solved.

**Comment by Suavesito-Olimpiada** [comment]
I also added access to `Symbols`.

**Comment by Suavesito-Olimpiada** [comment]
Fixed that `texstring` was not exported in the shared library.

File: src/texprintfsymbols [hunk 2]
```diff
@@ -1,13 +1,16 @@
+TEXPRINTF_SYMBOLS
 TEXPRINTF_LW
 TEXPRINTF_FONT
 TEXPRINTF_FCW
 TEXPRINTF_WCW
 TEXPRINTF_ERR
 texprintf
 stexprintf
+texstring
```
Comment (stevengj) [inline src/texprintfsymbols:9]: Whoops, sorry about that.

File: src/lexer.c [hunk 3]
```diff
@@ -52,6 +52,43 @@ void ListSymbols()
...
+	j=0;
+	while(Symbols[j].name)
+	{
+		str=Unicode2Utf8(Symbols[j].unicode);
+		if(IsCombiningMark(Symbols[j].unicode))
+		{
+			slen=(size_t)snprintf(NULL, 0, "%s %s:%s%s;", Symbols[j].name, p, str, p);
+			while (s < (ln+slen+1))
+				s*=2;
+			line=realloc(line, ln+slen+1);
```
Comment (stevengj) [inline src/lexer.c:74]: Shouldn't you be using `realloc(line, s)`, i.e. `s` is the new length?

File: src/texprintfsymbols (hunk 2 above)
Comment (Suavesito-Olimpiada) [reply src/texprintfsymbols:9]: No worries, I didn't notice in the other PR. Until testing the export of `TEXPRINTF_SYMBOLS` I got the error. :b

File: src/lexer.c (hunk 3 above)
Comment (Suavesito-Olimpiada) [reply src/lexer.c:74]: Yes.

File: src/lexer.c [hunk 4]
```diff
@@ -52,6 +52,43 @@ void ListSymbols()
...
+			snprintf(line+ln, slen+1, "%s:%s;", Symbols[j].name, str);
+			ln+=slen;
+		}
+		free(str);
+		j++;
+	}
+	free(p);
+	return line;
```
Comment (stevengj) [inline src/lexer.c:91]:
```suggestion
	return realloc(line, ln+1);
```
if you over-allocate to length `s` as suggested above.

Comment (Suavesito-Olimpiada) [reply src/lexer.c:91]: Done.

**Pull request merged**